client/asset/eth/eth.go

```diff
@@ -1909,11 +1909,15 @@ func (w *assetWallet) Redeem(form *asset.RedeemForm, feeWallet *assetWallet, non
 
 	gasLimit, gasFeeCap := g.Redeem*n, form.FeeSuggestion
 	originalFundsReserved := gasLimit * gasFeeCap
-	gasEst, err := w.estimateRedeemGas(w.ctx, secrets, contractVer)
-	if err != nil {
+
+	/* We could get a gas estimate via RPC, but this will reveal the secret key
+	   before submitting the redeem transaction. This is not OK for maker. This
+	   could be acceptable if any error from estimateRedeemGas is not fatal, and
+	   we redeem regardless (using hard-coded gas). Disable for now.
```

Comment:
> This could be acceptable if any error from estimateRedeemGas is not fatal, and we redeem regardless

I question whether even that is appropriate. If the risk is that the provider is in on the scame, then they could just deny our redemption broadcast.

Reply:
I agree with that.  I'll revise the comment so it doesn't suggest it as a possibility.

Reply:
I do still want to leave this code in case we one day have a privacy-preserving client wallet.🤞